### PR TITLE
add subcommand "statefulset" for "kubectl create"

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:
xref [kubernetes/kubectl#12](https://github.com/kubernetes/kubectl/issues/12)

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
/assign @mengqiy @pwittrock 

**Release note**:

```release-note
add subcommand "statefulset" for "kubectl create"
```

### CODE REVIEW

**Comment by cblecker** [comment]
/ok-to-test
Please note: We are currently in the 1.8 release code freeze and as such this PR will not be able to be merged until that lifts. Thanks for your contribution!

File: hack/make-rules/test-cmd-util.sh [hunk 1]
```diff
@@ -526,6 +531,7 @@ run_pod_tests() {
   kubectl delete configmap test-configmap --namespace=test-kubectl-describe-pod
   kubectl delete pdb/test-pdb-1 pdb/test-pdb-2 pdb/test-pdb-3 pdb/test-pdb-4 --namespace=test-kubectl-describe-pod
   kubectl delete namespace test-kubectl-describe-pod
+  kubectl delete statefulset test-sts --namespace=test-kubectl-describe-pod
```
Comment (zjj2wry) [inline hack/make-rules/test-cmd-util.sh]: seems you delete this after delete namespace.

Comment (dixudx) [reply hack/make-rules/test-cmd-util.sh]: Aha. Thanks for pointing out.

File: hack/make-rules/test-cmd-util.sh [hunk 2]
```diff
@@ -525,6 +530,7 @@ run_pod_tests() {
   kubectl delete secret test-secret --namespace=test-kubectl-describe-pod
   kubectl delete configmap test-configmap --namespace=test-kubectl-describe-pod
   kubectl delete pdb/test-pdb-1 pdb/test-pdb-2 pdb/test-pdb-3 pdb/test-pdb-4 --namespace=test-kubectl-describe-pod
+  kubectl delete statefulset test-sts --namespace=test-kubectl-describe-pod
```
Comment (zjj2wry) [inline hack/make-rules/test-cmd-util.sh]: you should put this line to L496, in L521 has check no pod logic so cause test failed

**Comment by dixudx** [comment]
@mengqiy @pwittrock Rebased. PTAL. Thanks.

File: pkg/kubectl/cmd/create_statefulset.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+	statefulSetExample = templates.Examples(i18n.T(`
+	# Create a new StatefulSet named my-sts that runs the busybox image.
+	kubectl create statefulset my-sts --image=busybox`))
+)
+
+// NewCmdCreateStatefulSet is a command to create a new StatefulSet.
+func NewCmdCreateStatefulSet(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "statefulset NAME --image=image [--dry-run]",
```
Comment (apelisse) [inline pkg/kubectl/cmd/create_statefulset.go]: @foxish should there be other flags there? initcontainer?

**Comment by apelisse** [comment]
Do we still want this?

**Pull request closed without merging**